org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ExecuteView.java

```diff
@@ -649,19 +655,19 @@ private String getDetectorNames() throws Exception {
     	return detectors;
 	}
 
-	private List<String> getMonitors() throws Exception {
+	private List<String> getMonitors(MonitorRole monitorRole) throws Exception {
 
 		final Collection<DeviceInformation<?>> scannables = cservice.getDeviceInformation();
 		final List<String> ret = new ArrayList<String>();
 		for (DeviceInformation<?> info : scannables) {
-			if (info.isActivated()) ret.add(info.getName());
+			if (info.isActivated() && info.getMonitorRole() == monitorRole) ret.add(info.getName());
 		}
 		return ret;
 	}
 
-	private String getMonitorNames() throws Exception {
+	private String getMonitorNames(MonitorRole monitorRole) throws Exception {
 
-		List<String> mons = getMonitors();
+		List<String> mons = getMonitors(monitorRole);
 		if (mons==null || mons.isEmpty()) return "None";

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)